src/components/dialogs/create-study-dialog/create-study-dialog.js

```diff
@@ -0,0 +1,333 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { useForm } from 'react-hook-form';
+import { Grid } from '@mui/material';
+import { useIntl } from 'react-intl';
+import React, { useCallback, useEffect, useState } from 'react';
+import UploadNewCase from '../commons/upload-new-case';
+import {
+    createCaseWithoutDirectoryElementCreation,
+    createStudy,
+    deleteCase,
+    getCaseImportParameters,
+} from '../../../utils/rest-api';
+import {
+    HTTP_CONNECTION_FAILED_MESSAGE,
+    HTTP_UNPROCESSABLE_ENTITY_STATUS,
+} from '../../../utils/UIconstants';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { useDispatch, useSelector } from 'react-redux';
+import ImportParametersSection from './importParametersSection';
+import { ElementType } from '../../../utils/elementType';
+import DirectorySelect from './directory-select';
+import { useNameCheck } from '../commons/use-name-check';
+import { keyGenerator } from '../../../utils/functions';
+import {
+    addUploadingElement,
+    removeUploadingElement,
+    setActiveDirectory,
+} from '../../../redux/actions';
+import {
+    createStudyDialogFormValidationSchema,
+    getCreateStudyDialogFormDefaultValues,
+} from './create-study-dialog-utils';
+import {
+    API_CALL,
+    CASE_FILE,
+    CASE_NAME,
+    CASE_UUID,
+    DESCRIPTION,
+    FORMATTED_CASE_PARAMETERS,
+    STUDY_NAME,
+} from '../../utils/field-constants';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+import CustomMuiDialog from '../custom-mui-dialog';
+import { ErrorInput, FieldErrorAlert, TextInput } from '@gridsuite/commons-ui';
+import StudyNamePrefilledInput from './study-name-prefilled-input';
+
+const MAX_FILE_SIZE_IN_MO = 100;
+const MAX_FILE_SIZE_IN_BYTES = MAX_FILE_SIZE_IN_MO * 1024 * 1024;
+
+const CreateStudyDialog = ({ open, onClose, providedExistingCase }) => {
+    const intl = useIntl();
+    const { snackError } = useSnackMessage();
+    const dispatch = useDispatch();
+
+    const [caseFileLoading, setCaseFileLoading] = useState(false);
+
+    const activeDirectory = useSelector((state) => state.activeDirectory);
+    const selectedDirectory = useSelector((state) => state.selectedDirectory);
+    const userId = useSelector((state) => state.user.profile.sub);
+
+    const createStudyFormMethods = useForm({
+        mode: 'onChange',
+        defaultValues: getCreateStudyDialogFormDefaultValues({
+            directory: activeDirectory,
+        }),
+        resolver: yupResolver(createStudyDialogFormValidationSchema),
+    });
+
+    const {
+        setValue,
+        formState: { isValid },
+        setError,
+        clearErrors,
+        getValues,
+        watch,
+    } = createStudyFormMethods;
+
+    // Constants
+    const formattedCaseParameters = getValues(FORMATTED_CASE_PARAMETERS);
+    const studyName = watch(STUDY_NAME);
```

Comment:
as discussed, we can not use only field in use-name-check, because we need isChecking returned from this custom hook..
so useNameCheck will be outside the formProvider of CustomMuiDialog..
I will use a async checking in the form schema in a next TS. this way we will not use useNameCheck